libs/design/tree/src/tree-item/tree-item.directive.ts

```diff
@@ -135,30 +138,51 @@ export class DaffTreeItemDirective {
   @HostListener('click')
   onClick() {
     if(this.node.hasChildren) {
-      this.toggleTree(this.node);
+      this.toggleTree();
     }
+  }
+
+  scrollIntoView() {
+    (<HTMLElement>this.elementRef.nativeElement).scrollIntoView();
+  }
+
+  /**
+   * Opens parent and parent of parent all the way to the root of the tree.
+   */
+  openAncestors() {
+    const openParent = (tree: DaffTreeUi<unknown>) => {
+      if (tree?.parent.parent === undefined) {
+        return;
+      }
+      tree.parent.open = true;
+      openParent(tree.parent);
+    };
+    openParent(this._node._treeRef);
+    (<Document>this.document).getElementById('tree-' + this.node._treeRef.parent.id).focus();
     this.treeNotifier.notify();
   }
 
   /**
    * Toggle the open state of the tree's parent.
    */
-  toggleParent(node: DaffTreeFlatNode) {
-    if(node._treeRef?.parent.parent === undefined) {
+  toggleParent() {
+    if(this.node._treeRef?.parent.parent === undefined) {
       return;
     }
-    node._treeRef.parent.open = !node._treeRef.parent.open;
-    (<Document>this.document).getElementById('tree-' + node._treeRef.parent.id).focus();
+    this.node._treeRef.parent.open = !this.node._treeRef.parent.open;
+    (<Document>this.document).getElementById('tree-' + this.node._treeRef.parent.id).focus();
```

Comment:
Do this without dom access. This likely requires a focus function on treeItemDirectives.